### ISSUE TITLE
Lint on "too large" unsafe blocks

### ISSUE THREAD

**Issue by Ticki** [Manishearth/rust-clippy#716]
For example

```rust
unsafe {
    some_completely_safe_function();
    some_unsafe_function();
}
```

should generate a warning, which suggests to use

```rust
some_completely_safe_function();
unsafe {
    some_unsafe_function();
}
```

instead, since you want to keep the unsafe blocks as small and restricted as possible, to make the distinction between safe and unsafe as clear as possible.

**Comment by llogiq** [comment]
I personally don't think the size of unsafe blocks actually matters that much in practice. If you have but one `unsafe` block in your code, you still need to check for all the preconditions/invariants that code needs to uphold Rust's guarantees, and then check *the whole module* to ensure those are met.

**Comment by Ticki** [comment]
Yeah, that's true. Unsafe blocks are fuzzy. I still think that safe variable assignments are preferable to have outside the unsafe block.

**Comment by llogiq** [comment]
I used to think the same, but after having written a few thousand lines of Rust I think people are sufficiently wary of using `unsafe` at all.

**Comment by Ticki** [comment]
The different isn't big, but it _is_ more idiomatic to put safe code _outside_ `unsafe` blocks.

**Comment by Manishearth** [comment]
Not necessarily. I have seen a lot of people scoping unsafe blocks to the actual scope of the unsafety; Servo does this too

**Comment by Ticki** [comment]
Oh, yeah. Variable declarations should certainly happen inside unsafe blocks to limit the scope, but assignments or function calls should be, when they can, outside the block.

**Comment by Manishearth** [comment]
No, I mean that the unsafe block includes code that maintains the invariants that makes the unsafe block actually safe

**Comment by Ticki** [comment]
Oh, yeah. That's correct. It can be quite hard to distinguish between them, and it is probably not worth it anyways, so I'll close this.
